assignment/scripts/3-music-collection.js

```diff
@@ -1 +1,87 @@
-console.log('***** Music Collection *****')
+console.log('***** Music Collection *****');
+
+// global variabls
+let collection = [];
+
+// functions
+let addToCollection = (title, artist, yearPublished) => {
+    let album = {
+        title: title,
+        artist: artist,
+        yearPublished: yearPublished,
+    };
+    collection.push(album);
+    return album;
+};
+
+let showCollection = (albumArray) => {
+    console.log(`Number of albums: ${albumArray.length}`);
+    for (let album of albumArray) {
+        console.log(
+            `${album.title} by ${album.artist},` +
+                ` released in ${album.yearPublished}`
+        );
+    }
+};
+
+let findByArtist = (artist) => {
+    let artistCollection = [];
+    for (let album of collection) {
+        if (album.artist.toLowerCase() === artist.toLowerCase()) {
+            artistCollection.push(album);
+        }
+    }
+    return artistCollection;
+};
+
+let search = (criteria) => {
+    let matchingAlbums = [];
+    for (let album of collection) {
+        let isMatching = Object.entries(criteria).reduce(
```

Comment:
Nice use of `Object.entries` and `reduce`! Could also have replaced reduce with `every`.